### PR TITLE
New placement algorithm & datasource

### PR DESCRIPTION
**Pull request by MaYohann**
https://github.com/sigrennesmetropole/cooperation_jn_solar/assets/91056006/dbdfb658-e272-47c9-ac51-46ca73435e90

https://github.com/sigrennesmetropole/cooperation_jn_solar/assets/91056006/d24f01ab-1ef5-46f8-af4d-3228ef584722

### CODE REVIEW

File: src/services/roofInteractionHelper.ts [hunk 1]
```diff
@@ -183,18 +218,26 @@ export function displayGridOnMap(
...
-export function displayRoofShape(
+export function displayRoofShape2d(
   rennesApp: RennesApp,
-  geojson: GeoJSONFeatureCollection
+  geojson: FeatureCollection
 ) {
+  cleanRoofShape2d(rennesApp)
   const roofLayer: GeoJSONLayer = rennesApp.layers.getByKey(
-    RENNES_LAYER.roofShape
+    'roofShape2d'
```
Comment (ismailsunni) [inline src/services/roofInteractionHelper.ts]: not using RENNES_LAYER ?

File: src/services/solarPanel.ts [hunk 2]
```diff
@@ -128,20 +125,20 @@ export function getPeakPower() {
...
   const positions: number[][] = []
-  for (let i = 0; i < solarPanelGrids.length; i++) {
-    const spg = solarPanelGrids[i]
-    const index = i
-
-    const center = getSolarPanelGridCenter(matrixGrid, spg)
+  let index: number = 0
+  let offset = HeightOffset
+  if (isRoofFlat()) {
+    offset += 0.2
```
Comment (ismailsunni) [inline src/services/solarPanel.ts]: Could you give a comment about the 0.2 value?

Comment (ismailsunni) [reply src/services/solarPanel.ts]: Is the same as `HeightOffset` above?

File: src/services/solarPanel.ts [hunk 3]
```diff
@@ -208,15 +184,9 @@ function isRoofFlat() {
...
-  }
   return selectedRoofInclinaison
 }
 
 export function getAzimuthSolarPanel(selectedRoofAzimuth: number) {
-  if (isRoofFlat()) {
-    return getNumberFromConfig('solar_panel.azimuth_on_flat_roof')
-  }
-  return selectedRoofAzimuth
+  return selectedRoofAzimuth - 3.47
```
Comment (ismailsunni) [inline src/services/solarPanel.ts:179]: Put also a comment for 3.47

**Review by ismailsunni — APPROVED** [review]
LGTM, minor comments only

File: src/services/solarPanel.ts (hunk 2 above)
Comment (MaYohann) [reply src/services/solarPanel.ts]: I genuinely do not know why it is the 0.2 and think this was a offset set by you. I assumed we experiences issues with rendering 3d panels on flat roof, so we need to put them slighly above the roof? :)

Comment (ismailsunni) [reply src/services/solarPanel.ts]: Hmm, the heightoffset is from me I guess. But in this part, you add more 0.2. Is that correct?

Comment (MaYohann) [reply src/services/solarPanel.ts]: The +0.2 did not come from me, i will test it to see if it's still mandatory before merging

**Pull request merged**